packages/react/src/Button/Button.tsx

```diff
@@ -6,26 +6,33 @@
 import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ButtonHTMLAttributes, ForwardedRef, PropsWithChildren } from 'react'
+import { Icon } from '../Icon'
 
 export type ButtonProps = {
+  /** An icon to add to the button. */
+  icon?: Function
+  /** The position of the icon. The default is after the label. */
+  iconPosition?: 'start' | 'only'
   /** The level of prominence. Use a primary button only once per page or section. */
   variant?: 'primary' | 'secondary' | 'tertiary'
 } & PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>
 
 export const Button = forwardRef(
   (
-    { children, className, type, disabled, variant = 'primary', ...restProps }: ButtonProps,
+    { children, className, disabled, icon, iconPosition, type, variant = 'primary', ...restProps }: ButtonProps,
     ref: ForwardedRef<HTMLButtonElement>,
   ) => {
     return (
       <button
         {...restProps}
         ref={ref}
         disabled={disabled}
-        className={clsx('ams-button', `ams-button--${variant}`, className)}
+        className={clsx('ams-button', `ams-button--${variant}`, `ams-button--icon-${iconPosition}`, className)}
```

Comment:
Do we need to render classes for ‘start’ and ‘end’? The stylesheets do not contain selectors for them.

We can do this instead:

```
iconPosition === 'only' && `.ams-button--icon-position-only`
```

And the class should contain `-position`.

This does remove our option to unit test the rendering of an icon at the start or end.

Reply:
I did previously but it can be removed now, also fixes the undefined.